src/mono/mono/mini/intrinsics.c

```diff
@@ -1007,6 +1007,54 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 			EMIT_NEW_BIALU_IMM (cfg, ins, OP_COMPARE_IMM, -1, dreg, 0);
 			EMIT_NEW_UNALU (cfg, ins, OP_ICGT, dreg, -1);
 			ins->type = STACK_I4;
+			return ins;
+		} else if (!strcmp (cmethod->name, "CreateSpan") && fsig->param_count == 1) {
+			MonoGenericContext* ctx = mono_method_get_context (cmethod);
+			g_assert (ctx);
+			g_assert (ctx->method_inst);
+			g_assert (ctx->method_inst->type_argc == 1);
+			MonoType* arg_type = ctx->method_inst->type_argv [0];
+			MonoType* t = mini_get_underlying_type (arg_type);
+			g_assert (!MONO_TYPE_IS_REFERENCE (t) && t->type != MONO_TYPE_VALUETYPE);
+
+			// The following relies on the previous instruction being an OP_MOVE. Specifically. one that
+			// is emitted from CEE_TDTOKEN as the last EMIT_NEW_TEMPLOAD, which has its inst_p1 set.
+			// Bail out and use the non-intrinsic variant if this is not satisfied.
+			MonoClassField* field = (MonoClassField*) args [0]->inst_p1;
+			if (args [0]->opcode != OP_VMOVE || !field || !field->type)
+					return NULL;
+
+			int alignment = 0;
+			int element_size = mono_type_size (t, &alignment);
+
+		#if G_BYTE_ORDER == G_LITTLE_ENDIAN
+			int swizzle = 1;
+		#else
+			int swizzle = element_size;
+		#endif
+
+			gpointer data_ptr = (gpointer)mono_field_get_rva (field, swizzle);
+			const int num_elements = mono_type_size (field->type, &alignment) / element_size;
+			const int obj_size = MONO_ABI_SIZEOF (MonoObject);
+			
+			MonoInst* span = mono_compile_create_var (cfg, fsig->ret, OP_LOCAL);
+			MonoInst* span_addr;
+			EMIT_NEW_TEMPLOADA (cfg, span_addr, span->inst_c0);
+
+			MonoClassField* field_ref = mono_class_get_field_from_name_full (span->klass, "_reference", NULL);
+			MonoInst* ptr_inst;
+			if (cfg->compile_aot) {
+				EMIT_NEW_SFLDACONST (cfg, ptr_inst, field);
```

Comment:
Will this work for these kinds of fields ? At runtime, the code in mono_resolve_patch_target_ext () gets executed to retrieve the field address, i.e. the
MONO_PATCH_INFO_SFLDA case.

Reply:
I see there is also `MONO_PATCH_INFO_RVA`, maybe it would be more appropriate.